### PR TITLE
CORE-8023 Renamed 'libs/http-rpc-client' module to 'libs/rest-client'

### PR DESCRIPTION
**Pull request by ben-millar**
### Overview:
Renamed the `libs/http-rpc-client` module to `libs/rest-client` and refactored the contents to replace `rpc` with `rest` in any classes defined within this module. This PR satisfies the [CORE-8023](https://r3-cev.atlassian.net/browse/CORE-8023) task as part of the broader initiative described in [CORE-7040](https://r3-cev.atlassian.net/browse/CORE-7040).

### Testing:
Local tests are passing both as part of the composite build (`./gradlew publishOSGiImage`), and as a standalone build (`./gradlew build`)

[CORE-8023]: https://r3-cev.atlassian.net/browse/CORE-8023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-7040]: https://r3-cev.atlassian.net/browse/CORE-7040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ package net.corda.applications.workers.rpc.http
 import net.corda.applications.workers.rpc.utils.AdminPasswordUtil.adminPassword
 import net.corda.applications.workers.rpc.utils.AdminPasswordUtil.adminUser
 import net.corda.httprpc.RestResource
-import net.corda.httprpc.client.HttpRpcClient
+import net.corda.httprpc.client.RestClient
 
 /**
  * Toolkit for HTTP RPC E2E tests execution
```
Comment (pmck91) [inline applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt]: just leaving a comment to make a note of this file

File: libs/http-rpc/rest-client/src/main/kotlin/net/corda/httprpc/client/RestClient.kt [hunk 2]
```diff
@@ -18,45 +18,45 @@ import java.util.concurrent.CopyOnWriteArraySet
 import kotlin.concurrent.scheduleAtFixedRate
 
 /**
- * [HttpRpcClient] is meant to run outside of Corda Node JVM and provide connectivity to a node using the HTTP-RPC
```
Comment (pmck91) [inline libs/http-rpc/rest-client/src/main/kotlin/net/corda/httprpc/client/RestClient.kt:21]: Reference to HTTP-RPC in comments

**Review by pmck91 — DISMISSED** [review]
LGTM to make a note of the two comments that need to be updated either in a new pr or here

**Review by pmck91 — APPROVED** [review]

**Review by mag1c-48 — APPROVED** [review]
LGTM - build only

File: applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt [hunk 3]
```diff
@@ -3,10 +3,10 @@ package net.corda.applications.workers.rpc.http
 import net.corda.applications.workers.rpc.utils.AdminPasswordUtil.adminPassword
 import net.corda.applications.workers.rpc.utils.AdminPasswordUtil.adminUser
 import net.corda.httprpc.RestResource
-import net.corda.httprpc.client.HttpRpcClient
+import net.corda.httprpc.client.RestClient
```
Comment (vkolomeyko) [inline applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt:6]: What about changing package name from `httprpc` to `rest`?

Comment (ben-millar) [reply applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt:6]: This should be covered by a future task; I'll merge this for now and follow up ASAP on the package rename.

Comment (ben-millar) [reply applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/http/TestToolkit.kt:6]: A note has been added in [CORE-8031](https://r3-cev.atlassian.net/browse/CORE-8031) to handle this.

**Pull request merged**